agent/containermetadata/write_metadata_windows.go

```diff
@@ -0,0 +1,79 @@
+// +build windows
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package containermetadata
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/aws/amazon-ecs-agent/agent/api"
+	"github.com/aws/amazon-ecs-agent/agent/engine/dockerclient"
+)
+
+const (
+	mountPoint                     = `C:\ProgramData\Amazon\ECS\metadata`
+	ContainerMetadataClientVersion = dockerclient.Version_1_24
+)
+
+// createMetadataFile initializes the metadata file
+func createMetadataFile(metadataDirectoryPath string) error {
+	metadataFilePath := filepath.Join(metadataDirectoryPath, metadataFile)
+	//	err := ioutil.WriteFile(metadataFilePath, nil, readOnlyPerm)
+	file, err := os.Create(metadataFilePath)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+	err = file.Chmod(readOnlyPerm)
```

Comment:
I think you're trying to make it rw for the agent, but ro for the task containers.  It doesn't appear that this is effective on Windows.  If you look at [the `Chmod` source](https://github.com/golang/go/blob/release-branch.go1.7/src/syscall/syscall_windows.go#L527-L545), it looks like it only sets or clears the `FILE_ATTRIBUTE_READONLY` bit; in this case it will end up clearing the readonly attribute: https://play.golang.org/p/nMslTBgXgL.